src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -214,12 +218,36 @@ private void LoadCustomProviderFunctions(List<FunctionMetadata> functionMetadata
 
         private void AddMetadataFromCustomProviders(IEnumerable<IFunctionProvider> functionProviders, List<FunctionMetadata> functionMetadataList)
         {
-            _logger.ReadingFunctionMetadataFromProvider(_metadataProviderName);
+            _logger.ReadingFunctionMetadataFromProvider(MetadataProviderName);
 
             var functionProviderTasks = new List<Task<ImmutableArray<FunctionMetadata>>>();
+
             foreach (var functionProvider in functionProviders)
             {
-                functionProviderTasks.Add(functionProvider.GetFunctionMetadataAsync());
+                var getFunctionMetadataFromProviderTask = functionProvider.GetFunctionMetadataAsync();
+                var delayTask = Task.Delay(TimeSpan.FromSeconds(MetadataProviderTimeoutInSeconds));
+
+                var completedTask = Task.WhenAny(getFunctionMetadataFromProviderTask, delayTask).ContinueWith(t =>
+                {
+                    if (t.Result == getFunctionMetadataFromProviderTask)
+                    {
+                        if (getFunctionMetadataFromProviderTask.IsFaulted)
+                        {
+                            // Task completed but with an error
+                            _logger.LogWarning($"Failure in retrieving metadata from '{functionProvider.GetType().FullName}': {getFunctionMetadataFromProviderTask.Exception?.Flatten().ToString()}");
```

Comment:
Pass the values as arguments into your log call:

Example:
``` csharp
_logger.LogWarning("Failure in retrieving metadata from '{typeName}': {exception}", functionProvider.GetType().FullName, getFunctionMetadataFromProviderTask.Exception?.Flatten());
````

Reply:
Also, should this be an error instead?

Reply:
Removed the logging calls in the latest iteration (throwing an exception instead)